File: projetos-fenix-para-estudo/t2ti-erp-fenix-banco-agencia-pessoa-completos/fontes/backend/node/t2ti_erp_fenix/src/model/transiente/retorno_json_erro.js
/// classe transiente que armazena o objeto do erro que será retornado para o cliente
class RetornoJsonErro {
    constructor(parametros) {
        this.status = parametros.codigo;
        this.message = parametros.mensagem;
        switch (this.status) {
            case 400:
                this.error = "Bad Request";
                break;
            case 404:
                this.error = "Not Found";
                break;
            case 500:
                this.error = "Internal Server Error";
                break;
            default:
                this.error = "Erro não mapeado";
                break;
        }
        if (parametros.erro != null) {
            this.message = this.message + " - Exceção: " + parametros.erro.message;
            this.trace = parametros.erro.stack;
        }
    }
}

module.exports = RetornoJsonErro;

File: projetos-fenix-para-estudo/t2ti-erp-fenix-servidores-provider-inicio/fontes/frontend/provider/lib/src/model/retorno_json_erro.dart
/// classe transiente para armazenar o retorno de erros que vem do servidor
class RetornoJsonErro {
  String status;
  String error;
  String message;
  String trace;

  RetornoJsonErro({this.status, this.error, this.message});

  RetornoJsonErro.fromJson(Map<String, dynamic> jsonDados) {
    status = jsonDados['status'];
    error = jsonDados['error'];
    message = jsonDados['message'];
    trace = jsonDados['trace'];
  }

  Map<String, dynamic> get toJson {
    Map<String, dynamic> jsonDados = new Map<String, dynamic>();
    jsonDados['status'] = this.status;
    jsonDados['error'] = this.error;
    jsonDados['message'] = this.message;
    jsonDados['trace'] = this.trace;
    return jsonDados;
  }
}